### PR TITLE
Removing JEC branches from bigNtuples

### PR DESCRIPTION
**Pull request by dzuolo**
As suggested by @francescobrivio after the last meeting I removed from the bigNtuples the branches containing the JEC variated jets. This is done in KLUB (https://github.com/LLRCMS/KLUBAnalysis/pull/242) since the last stages of the non-resonant analysis when having new JECs was necessary after a mistake was found in the txt files used for the big ntuple production. I tried to run on some MC files and I observed that the other jets branches are filled correctly. @portalesHEP if you could have a look too that would be great :)
P.S. We will use only the regrouped JECs as for the non-resonant analysis.

### CODE REVIEW

**Comment by portalesHEP** [comment]
Actually (probably) related to previous PR mostly: I'm running the framework for tests, and noticed these warning that I missed last time:
```
%MSG-e L1PrefireWeightProducer:  L1PrefiringWeightProducer:prefiringweight@ctor  22-Feb-2022 17:12:03 CET pre-events
Photon map not found. All photons prefiring weights set to 0. 
%MSG
%MSG-e L1PrefireWeightProducer:  L1PrefiringWeightProducer:prefiringweight@ctor  22-Feb-2022 17:12:03 CET pre-events
Jet map not found. All jets prefiring weights set to 0. 

``` 
should that be worrying?

**Comment by dzuolo** [comment]
I also noticed that warning and honestly I do not know. According to the prefiring twiki page the tool can be used out of the box for CMSSW > 10_6_26 so I did not give much importance to it. Also: this should be related to the object weight we do not use since we take the event weight. If we want to be on the safe side we can get in touch with people working on this?

**Comment by francescobrivio** [comment]
> Actually (probably) related to previous PR mostly: I'm running the framework for tests, and noticed these warning that I missed last time:
> 
> ```
> %MSG-e L1PrefireWeightProducer:  L1PrefiringWeightProducer:prefiringweight@ctor  22-Feb-2022 17:12:03 CET pre-events
> Photon map not found. All photons prefiring weights set to 0. 
> %MSG
> %MSG-e L1PrefireWeightProducer:  L1PrefiringWeightProducer:prefiringweight@ctor  22-Feb-2022 17:12:03 CET pre-events
> Jet map not found. All jets prefiring weights set to 0. 
> ```
> 
> should that be worrying?

@dzuolo @portalesHEP yes let's get in touch with the developers, but maybe @camendola knows if prefiring weights even exist for photons?

**Comment by portalesHEP** [comment]
> As suggested by @francescobrivio after the last meeting I removed from the bigNtuples the branches containing the JEC variated jets. This is done in KLUB ([LLRCMS/KLUBAnalysis#242](https://github.com/LLRCMS/KLUBAnalysis/pull/242)) since the last stages of the non-resonant analysis when having new JECs was necessary after a mistake was found in the txt files used for the big ntuple production. I tried to run on some MC files and I observed that the other jets branches are filled correctly. @portalesHEP if you could have a look too that would be great :) P.S. We will use only the regrouped JECs as for the non-resonant analysis.

Regarding this PR, all seems fine. I did run on my side on some DY MC as well and can confirm that nothing unexpected seems to happen (and the test file size was reduced by ~30% which is pretty nice !)

**Comment by dzuolo** [comment]
This is good news! Thanks Louis!

**Comment by dzuolo** [comment]
And the doubt is solved

Hello,

Yes there's no photon/jet prefiring maps for 2018 as it is supposed to be negligible.

Best,

Laurent

**Comment by francescobrivio** [comment]
> And the doubt is solved
> 
> Hello,
> 
> Yes there's no photon/jet prefiring maps for 2018 as it is supposed to be negligible.
> 
> Best,
> 
> Laurent

@portalesHEP can you confirm if you get this errors in 2018 only or also in 2016/2017?

File: NtupleProducer/python/triggers_92X.py [hunk 1]
```diff
@@ -574,13 +574,17 @@
         HLT = cms.string("HLT_PFMETNoMu120_PFMHTNoMu120_IDTight_v"),
         path1 = cms.vstring ("",""),
         path2 = cms.vstring (""),
+	path3 = cms.vstring(""),
+	path4 = cms.vstring(""),
```
Comment (francescobrivio) [inline NtupleProducer/python/triggers_92X.py]: The intendation is wrong.

File: NtupleProducer/python/triggers_92X.py [hunk 2]
```diff
@@ -574,13 +574,17 @@
...
         leg1 = cms.int32(999),
         leg2 = cms.int32(999),
         ),
     cms.PSet (
         HLT = cms.string("HLT_PFMETNoMu120_PFMHTNoMu120_IDTight_PFHT60_v"),
         path1 = cms.vstring ("",""),
         path2 = cms.vstring (""),
+	path3 = cms.vstring(""),
+	path4 = cms.vstring(""),
```
Comment (francescobrivio) [inline NtupleProducer/python/triggers_92X.py]: Here as well

**Comment by portalesHEP** [comment]
> @portalesHEP can you confirm if you get this errors in 2018 only or also in 2016/2017?

Yes, I only checked 2018

**Comment by dzuolo** [comment]
I checked all the 4 periods and the warning was present only in 2018.

**Comment by francescobrivio** [comment]
> > As suggested by @francescobrivio after the last meeting I removed from the bigNtuples the branches containing the JEC variated jets. This is done in KLUB ([LLRCMS/KLUBAnalysis#242](https://github.com/LLRCMS/KLUBAnalysis/pull/242)) since the last stages of the non-resonant analysis when having new JECs was necessary after a mistake was found in the txt files used for the big ntuple production. I tried to run on some MC files and I observed that the other jets branches are filled correctly. @portalesHEP if you could have a look too that would be great :) P.S. We will use only the regrouped JECs as for the non-resonant analysis.
> 
> Regarding this PR, all seems fine. I did run on my side on some DY MC as well and can confirm that nothing unexpected seems to happen (and the test file size was reduced by ~30% which is pretty nice !)

I ran on 5k event of 2018 DY and I see an even larger reduction! so that's great!

**Comment by dzuolo** [comment]
if you are all happy with this PR we can merge it. I will have a look at KLUB to include the total jet uncertainty which was previously taken from bigntuples.

**Comment by francescobrivio** [comment]
> if you are all happy with this PR we can merge it. I will have a look at KLUB to include the total jet uncertainty which was previously taken from bigntuples.

+1
but I let @portalesHEP merge it!

**Pull request merged**